### PR TITLE
allow setting initial value of `gr.Dropdown` to `None` to designate that no value should be initially selected

### PR DESCRIPTION
**Pull request by abidlabs**
Context: [internal link](https://huggingface.slack.com/archives/C07N8RBSR39/p1728709783612359)

Test with:

```py
import gradio as gr

with gr.Blocks() as demo:
    gr.Dropdown(["abc", "def"])
    gr.Dropdown(["abc", "def"], multiselect=True)
    gr.Dropdown(["abc", "def"], value=None)  # only this behavior is changed
    gr.Dropdown(["abc", "def"], value="abc")
    gr.Dropdown(["abc", "def"], value="def")

demo.launch()
```

### CODE REVIEW

**Review by hysts — APPROVED** [review]
I tested this and LGTM! Thanks!

File: test/components/test_dropdown.py [hunk 1]
```diff
@@ -22,12 +22,14 @@ def test_component_functions(self):
...
         assert dropdown.preprocess("a") == 0
         assert dropdown.preprocess("b") == 1
+        assert dropdown.value == "a"
         with pytest.raises(gr.Error):
             dropdown.preprocess("c")
 
         dropdown = gr.Dropdown(choices=["a", "b"], type="index", multiselect=True)
         assert dropdown.preprocess(["a"]) == [0]
         assert dropdown.preprocess(["a", "b"]) == [0, 1]
+        assert dropdown.value is None
```
Comment (whitphx) [inline test/components/test_dropdown.py:32]: I wonder this default value `None` for multiselect makes sense over `[]`. From users' perspective, it's more convenient if the value type is consistent (e.g. for `multiselect=True`, the value should always be of type `list` and never `None`.)
In other words, I think we should think of the good default behavior for the (`value=FIRST_SELECT`, `multiselect=True`) case.

I think, the default value for each option combination is following:
|         | `multiselect=False` | `multiselect=True` |
|---------|----------------------|---------------------|
| `value=FIRST_CHOICE` | `choices[0][1]` ✅ | *What does this mean?* Now this is `None` but `[]` makes more sense? |
| `value=None` | `None` ✅ | `None` (`None` for multiselect may not make sense but it's ok as the user specifies it.) |
| `value=[]` | `[]` (`[]` for singleselect may not make sense but it's ok as the user specifies it.) | `[]` ✅ |

One suggestion is, to rename `FIRST_CHOICE` to something just like `DEFAULT_VALUE` and set the default value as `choice[0][1]` for `multiselect=False` and `[]` for `multiselect=True`.

Comment (abidlabs) [reply test/components/test_dropdown.py:32]: Nice suggestion, will make this change:

>  rename FIRST_CHOICE to something just like DEFAULT_VALUE and set the default value as choice[0][1] for multiselect=False and [] for multiselect=True

Comment (abidlabs) [reply test/components/test_dropdown.py]: Made the change here: https://github.com/gradio-app/gradio/pull/9699/commits/7fd47b8bf9cfc8cf048e3f423f537406863b4b88

**Review by whitphx — APPROVED** [review]
LGTM!

File: test/components/test_dropdown.py (hunk 1 above)
Comment (whitphx) [reply test/components/test_dropdown.py]: Thanks!

**Comment by abidlabs** [comment]
Thanks @hysts and @whitphx for the reviews!

**Pull request merged**